pantheon-bundle/src/test/java/com/redhat/pantheon/servlet/assets/ImageServletFilterTest.java

```diff
@@ -0,0 +1,57 @@
+package com.redhat.pantheon.servlet.assets;
+
+import com.redhat.pantheon.conf.GlobalConfig;
+import org.apache.sling.testing.mock.sling.junit5.SlingContext;
+import org.apache.sling.testing.mock.sling.junit5.SlingContextExtension;
+import org.apache.sling.testing.mock.sling.servlet.MockRequestDispatcherFactory;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.RequestDispatcher;
+
+import java.util.Base64;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
+
+@ExtendWith({SlingContextExtension.class, MockitoExtension.class})
+class ImageServletFilterTest {
+
+    SlingContext sc = new SlingContext();
+
+    @Mock
+    FilterChain filterChain;
+
+    @Mock
+    MockRequestDispatcherFactory requestDispatcherFactory;
+
+    @Test
+    void doFilter() throws Exception {
+        // Given
+        String imagePath = "/path/to/my/image.png";
+        sc.build()
+                .resource(imagePath)
+                .commit();
+        String imageUrl = Base64.getUrlEncoder().encodeToString(imagePath.getBytes());
+        RequestDispatcher requestDispatcher = mock(RequestDispatcher.class);
+        sc.request().setRequestDispatcherFactory(requestDispatcherFactory);
+        sc.request().setPathInfo(GlobalConfig.IMAGE_PATH_PREFIX + "/" + imageUrl);
+        when(requestDispatcherFactory.getRequestDispatcher(anyString(), any())).thenReturn(requestDispatcher);
+        ImageServletFilter filter = new ImageServletFilter();
+
+        // When
+        filter.init(mock(FilterConfig.class));
+        filter.doFilter(sc.request(), sc.response(), filterChain);
+
+        // Then
+        verify(requestDispatcherFactory).getRequestDispatcher(eq(imagePath), any());
+        verify(requestDispatcher).forward(any(), any());
```

Comment:
Looks good, I think this is just verifying that your filter forwards the request when appropriate, right? Might be worth it to include a negative test - ensure that your filter doesn't interfere in a request that doesn't start with `IMAGE_PATH_PREFIX`.

Reply:
:+1: I can add that

Reply:
I think I read this wrong the first time.
I can't exactly test for that (at least not in a unit test) because that falls outside the responsibility of the filter. The filter assumes that it is being invoked. I would need to test the servlet framework class parsing the url and constructing the filter chain to make sure this filter is only invoked when the path matches.

Reply:
Ok, I'll buy that argument.